javascript/app-framework/src/types.ts

```diff
@@ -157,3 +157,7 @@ export enum ResourceTreeViewMode {
 	Collapsed = 'collapsed',
 	Hidden = 'hidden',
 }
+
+export interface ApplicationSettings {
```

Comment:
TODO: make this empty so there are no predefined settings, just a mechanism for declaring them on the DataShaperApp component. The built-in webapp will include `darkMode` as a functional example.